elixir/apps/api/lib/api/device/channel.ex

```diff
@@ -80,15 +80,20 @@ defmodule API.Device.Channel do
   end
 
   @impl true
-  def handle_in("list_relays", %{"resource_id" => resource_id}, socket) do
+  def handle_in(
+        "get_gateway_and_list_relays",
+        %{"resource_id" => resource_id, "connected_gateway_ids" => connected_gateway_ids},
+        socket
+      ) do
     with {:ok, resource} <- Resources.fetch_resource_by_id(resource_id, socket.assigns.subject),
          # :ok = Resource.authorize(resource, socket.assigns.subject),
          {:ok, [_ | _] = relays} <- Relays.list_connected_relays_for_resource(resource) do
       reply =
         {:ok,
          %{
            relays: Views.Relay.render_many(relays, socket.assigns.subject.expires_at),
-           resource_id: resource_id
+           resource_id: resource_id,
+           gateway_id: gateway.id
```

Comment:
We also can send `gateway_addresses: [ipv4, ipv6]` so that client can use it as additional direct ICE candidates.